expaghetti/parser.lua

```diff
@@ -124,18 +151,31 @@ local function parser(expr, flags,
 		end
 	end
 
-	if isGroup and not hasGroupClosed then
+	if isGroup and not hasGroupClosed and not isAlternate then
 		return false, errorsEnum.unterminatedGroup
 	end
 
-	return tree, index
+	return tree, index, hasGroupClosed
```

Comment:
It could be `(isGroup and hasGroupClosed or nil)` to prevent a third parameter when not needed, however it can already be nil indirectly and if it was nil during the loops it would set the variable to false N times, making it a little bit less performatic.